gtk4/src/snapshot.rs

```diff
@@ -15,16 +29,25 @@ impl Snapshot {
             let border_color_ptr: Vec<gdk::ffi::GdkRGBA> =
                 border_color.iter().map(|c| *c.to_glib_none().0).collect();
             ffi::gtk_snapshot_append_border(
-                self.to_glib_none().0,
+                self.as_ref().to_glib_none().0,
                 outline.to_glib_none().0,
                 border_width,
                 border_color_ptr.as_ptr() as *const _,
             )
         }
     }
 
-    #[doc(alias = "gtk_snapshot_push_debug")]
-    pub fn push_debug(&self, message: &str) {
-        unsafe { ffi::gtk_snapshot_push_debug(self.to_glib_none().0, message.to_glib_none().0) }
+    fn push_debug(&self, message: &str) {
+        unsafe {
+            ffi::gtk_snapshot_push_debug(self.as_ref().to_glib_none().0, message.to_glib_none().0)
+        }
     }
 }
+
+impl AsRef<Snapshot> for gdk::Snapshot {
+    fn as_ref(&self) -> &Snapshot {
+        self.downcast_ref().unwrap()
```

Comment:
This is technically wrong. `GdkSnapshot` is actually a different type than `GtkSnapshot`

Reply:
it is the parent type of GtkSnapshot https://docs.gtk.org/gtk4/class.Snapshot.html#hierarchy

Reply:
Yes, i.e. not every `GdkSnapshot` is a `GtkSnapshot` so the above can fail. Just that **for now** there's only a single subclass of `GdkSnapshot`.